pkg/controller/controller_instance.go

```diff
@@ -1389,10 +1390,31 @@ func (c *controller) prepareRequestHelper(instance *v1beta1.ServiceInstance, ser
 	rh.ns = ns
 
 	if setInProgressProperties {
-		parameters, parametersChecksum, rawParametersWithRedaction, err := prepareInProgressPropertyParameters(
+		// If the current operation is already set, we need to ignore the spec and read parameters from the status
+		var specParameters *runtime.RawExtension
+		var planExternalName string
+		var planExternalID string
+		var userInfo *v1beta1.UserInfo
+
+		if instance.Status.CurrentOperation == "" {
```

Comment:
@kibbles-n-bytes after making this change, I feel like it would be better to always have a `CurrentOperation` set before preparing an OSB request. In other words, instead of explicitly mitigating a potential issue with retries here, we just reorder logic in controller:
1. First record the intent in the status
2. Then use the recorded status as a source for OSB requests.

What do you think?